burr/core/application.py

```diff
@@ -2207,10 +2207,14 @@ def with_state_persister(
         """
         if on_every != "step":
             raise ValueError(f"on_every {on_every} not supported")
-        if hasattr(persister, 'initialize') and not persister.is_initialized():
-            raise RuntimeError(
-            "Uninitialized persister. Make sure to call .initialize() before passing it to the ApplicationBuilder."
-            )
+        
+        # Check if 'is_initialized' exists and whether it returns False, indicating the persister is uninitialized
+        if hasattr(persister, 'is_initialized'):
+            if not persister.is_initialized():
+                raise RuntimeError(
+                    "RuntimeError: Uninitialized persister. Make sure to call .initialize() before passing it to the ApplicationBuilder"
+                )
+        # If the persister is valid and initialized, add it to lifecycle adapters
```

Comment:
Sorry I think I was incorrect. 

1. Everyone should be inheriting from the base. So even custom persisters will have the `is_initialized` method.
2. Instead we want to catch the `not implementederror`.  

So something like this:

```suggestion
        try:
            if not persister.is_initialized():
                raise RuntimeError(
                    "RuntimeError: Uninitialized persister. Make sure to call .initialize() before passing it to the ApplicationBuilder"
                )
        except NotImplementedError:
             pass
        # If the persister is valid and initialized (or hasn't implemented `is_initialized()`), add it to lifecycle adapters
```

Reply:
we should then have a unit test for this case where a persister has not implemented the method.